translate-c: no align available for opaque types

```c
struct opaque;
struct opaque_2;

static inline void function(struct opaque *opaque) {
	struct opaque_2 *cast = (struct opaque_2 *)opaque;
}
```

```
/home/tdeo/test.zig:5:83: error: no align available for type 'struct_opaque_2'
    var cast: ?*struct_opaque_2 = @ptrCast(?*struct_opaque_2, @alignCast(@alignOf(struct_opaque_2), opaque_1));
```

Most of the functions in the Wayland headers cast their arguments to wl_proxy.

--- a/src-self-hosted/translate_c.zig
+++ b/src-self-hosted/translate_c.zig
@@ -2735,6 +2735,9 @@ fn transCPtrCast(
     if (ZigClangType_isVoidType(qualTypeCanon(child_type))) {
         // void has 1-byte alignment, so @alignCast is not needed
         try ptrcast_node.params.push(expr);
+    } else if (typeIsOpaque(rp.c, qualTypeCanon(child_type), loc)) {
+        // For opaque types a ptrCast is enough
+        try ptrcast_node.params.push(expr);
     } else {
         const aligncast_node = try transCreateNodeBuiltinFnCall(rp.c, "@alignCast");
         const alignof_node = try transCreateNodeBuiltinFnCall(rp.c, "@alignOf");

--- a/test/translate_c.zig
+++ b/test/translate_c.zig
@@ -3,6 +3,21 @@ const builtin = @import("builtin");
 
 pub fn addCases(cases: *tests.TranslateCContext) void {
     /////////////// Cases that pass for both stage1/stage2 ////////////////
+    cases.add("simple ptrCast for casts between opaque types",
+        \\struct opaque;
+        \\struct opaque_2;
+        \\void function(struct opaque *opaque) {
+        \\    struct opaque_2 *cast = (struct opaque_2 *)opaque;
+        \\}
+    , &[_][]const u8{
+        \\pub const struct_opaque = @OpaqueType();
+        \\pub const struct_opaque_2 = @OpaqueType();
+        \\pub export fn function(arg_opaque_1: ?*struct_opaque) void {
+        \\    var opaque_1 = arg_opaque_1;
+        \\    var cast: ?*struct_opaque_2 = @ptrCast(?*struct_opaque_2, opaque_1);
+        \\}
+    });
+
     cases.add("simple function prototypes",
         \\void __attribute__((noreturn)) foo(void);
         \\int bar(void);

